n2v/inverter.py

```diff
@@ -220,38 +217,92 @@ def invert(self, method,
+                    Becomes attribute of inverter -> inverter.lambda_reg
+                opt_tol: float
+                    tol for scipy.optimize.minimize
+            return:
+                the result are stored in self.v_opt
+
+        mRKS
+        the modified Ryabinkin-Kohut-Staroverov method:
+        ----------------------
+            parameters:
+                maxiter: int
+                    same as opt_max_iter
+                vxc_grid: np.ndarray of shape (3, num_grid_points), opt
+                    When this is given, the final result will be represented
+                v_tol: float, opt
+                    convergence criteria for vxc Fock matrices.
+                    default: 1e-4
+                D_tol: float, opt
+                    convergence criteria for density matrices.
+                    default: 1e-7
+                eig_tol: float, opt
+                    convergence criteria for occupied eigenvalue spectrum.
+                    default: 1e-4
+                frac_old: float, opt
+                    Linear mixing parameter for current vxc and old vxc.
+                    If 0, no old vxc is mixed in.
+                    Should be in [0,1)
+                    default: 0.5.
+                init: string or psi4.core.Wavefunction, opt
+                    Initial guess method.
+                    default: "SCAN"
+                    1) If None, input wfn info will be used as initial guess.
+                    2) If "continue" is given, then it will not initialize
+                    but use the densities and orbitals stored. Meaningly,
+                    one can run a quick WY calculation as the initial
+                    guess. This can also be used to user speficified
+                    initial guess by setting Da, Coca, eigvec_a.
+                    3) If it's not continue, it would be expecting a
+                    method name string that works for psi4. A separate psi4 calculation
+                    would be performed.
+            returns:
+                all are np.ndarray of shape (num_grid_points)
+                vxc, vxchole, ebarKS, ebarWF, taup_rho_WF, taup_rho_KS
+                in eqn vxc = vxchole + ebarKS - ebarWF + taup_rho_WF - taup_rho_KS.
+                Check the original paper for each component.
         """
 
         self.lam = lam
-        self.lambda_reg = reg
         self.generate_components(guide_potential_components)
 
         if method.lower() == "wuyang":
-            self.wuyang(opt_method, opt_max_iter, opt_tol)
+            self.wuyang(opt_max_iter, **opt)
         elif method.lower() == "zmp":
-            self.zmp_with_scf(lam, opt_max_iter, opt_tol)
+            self.zmp_with_scf(lam, opt_max_iter)
         elif method.lower() == "mrks":
-            pass
+            if guide_potential_components[0] != "hartree":
+                print("The guide potential is changed to v_hartree.")
+                self.generate_components(["hartree"])
```

Comment:
Can we set this restriction along line 294? I think it would be weird to have the potentials to be computed twice with in a case like this: `guide_potential_components = ['svwn', 'hartree']`.
Not sure what is the best option. Another solution is to move the `self.gen_components` inside each `if method` and force `hartree` guide for the mrks method. 
If hartree is also forced. We can get rid of the exception on line 338 of rmks.